tests/test_device_attach_notification_nordy/src/test_device_attach_notification_nordy.xc

```diff
@@ -0,0 +1,180 @@
+// Copyright 2016-2025 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+#include <xs1.h>
+#include <print.h>
+#include <stdio.h>
+#include "xud.h"
+#include "platform.h"
+#include "xud_shared.h"
+
+#define XUD_EP_COUNT_OUT    2
+#define XUD_EP_COUNT_IN     2
+
+#ifndef TEST_EP_NUM
+#define TEST_EP_NUM         1
+#endif
+
+#define PKT_LENGTH          10
+
+#ifndef XUD_TEST_SPEED
+#error XUD_TEST_SPEED not defined
+#endif
+
+/* Endpoint type tables */
+XUD_EpType epTypeTableOut[XUD_EP_COUNT_OUT] = {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL | XUD_STATUS_ENABLE};
+XUD_EpType epTypeTableIn[XUD_EP_COUNT_IN] =   {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL | XUD_STATUS_ENABLE};
+
+#pragma unsafe arrays
+testResult_t TestEpBusUpdate_Tx(XUD_ep ep_out0, XUD_ep ep_in, int epNum, int pktLen, t_runMode runMode)
+{
+    XUD_Result_t result;
+    XUD_BusSpeed_t busSpeed;
+    unsigned length;
+
+    unsigned char bufferRx[1024];
+    unsigned char bufferTx[1024];
+    unsigned char bufferTxBad[1024];
+
+    set_core_fast_mode_on();
+
+    /* Prepare packet */
+    for(int i = 0; i < pktLen; i++)
+    {
+        bufferTx[i] = i;
+        bufferTxBad[i] = 0xBA;
+    }
+
+    /* At this point (due to test packet sequencing and code ordering) that XUD has already sent
+     * a bus reset notification. This packet should never make it to the host */
+    result = XUD_SetBuffer(ep_in, bufferTxBad, pktLen);
```

Comment:
Shouldn't we do a 
if(result != XUD_RES_RST)
        return FAIL_UNEXPECTED_STATUS;
here?

Reply:
Will fix